notebooks/hierarchical_clustering.livemd

```diff
@@ -0,0 +1,217 @@
+# Hierarchical Clustering
+
+```elixir
+app_root = Path.join(__DIR__, "..")
+
+Mix.install(
+  [
+    {:kino, "~> 0.10.0"},
+    {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
+    {:scholar, path: app_root}
+  ],
+  config_path: Path.join(app_root, "config/config.exs"),
+  lockfile: Path.join(app_root, "mix.lock")
+)
+```
+
+## Introduction
+
+```elixir
+defmodule Kino.CanvasDendrogram do
+  use Kino.JS
+
+  def new(graph), do: Kino.JS.new(__MODULE__, graph)
+
+  asset "main.js" do
```

Comment:
I've kept this out of the sidebar for the moment. Despite the LOC, there's not much here except the plotting logic. See also the commented out section of the moduledoc in `Scholar.Cluster.Hierarchical`.

Reply:
Yeah, I have been long thinking it would be nice if Scholar already included visualizations for several of its modules, so I am :+1: for getting started with this trend. Just make sure to render it as a tabbed output, so we also have access to the "Raw" data structure.

Reply:
It may be worth keeping it as is (not visible in the sidebar) for now and making available in another PR. I looked at a few other guide notebooks in the repo and they're more detailed. This one doesn't even show the clusters, just the raw dendrogram. Typically a dendrogram plot will assign a color to each cluster.

Reply:
Yup, we can postpone adding it for sure. But to be clear, I was thinking about moving this Kino module itself into Scholar and declaring Kino an optional dependency. The goal is that many of our modules could come with built-in visualization, but that's a separate convo (hence let's postpone it). :)

Reply:
Btw. it should be `KinoScholar.CanvasDendrogram` or `Scholar.Kino.CanvasDendrogram`, the `Kino.` prefix is reserved for "official" kino packages, since for those we make sure there are no conflicts (and there are historic reasons :).

Reply:
[5589061](https://github.com/elixir-nx/scholar/pull/187/commits/5589061229fe244bc9ff3049e17d43e4cdad4a7c)